src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,136 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data_names,dim_names,coord_names=None,attr_names=None):
+        """Extracts metadata from a SmartRedis dataset and adds it to
+        dataformat specific fieldnames 
+
+        :param ds: a SmartRedis Dataset with metadata added to Xarray 
+        specific field names 
+        :type ds: SmartRedis Dataset
+        :param data_names: variable data field name
+        :type data_names: list or string 
```

Comment:
I think we want to put ``list[str] or str`` for these types.  Please look throughout the code for specifying the type inside of the list and using consistent type names.  ``client.py`` is a good place to look for examples.